core/include/kernel/pm.h

```diff
@@ -30,6 +30,12 @@
 #define PM_HINT_PLATFORM_STATE_MASK	GENMASK_32(31, 16)
 #define PM_HINT_PLATFORM_STATE_SHIFT	U(16)
 
+#define PM_HINT_STATE(x)		((x) & ~PM_HINT_PLATFORM_STATE_MASK)
+#define PM_HINT_PLATFORM_STATE(x) \
+	(((x) & PM_HINT_PLATFORM_STATE_MASK) >> PM_HINT_PLATFORM_STATE_SHIFT)
+
+#define PM_HINT_IS_STATE(x, name) (!!((x) & PM_HINT_ ## name ## _STATE))
```

Comment:
Can remove `!!` operation, `((x) & PM_HINT_ ## name ## _STATE)` is fine.

Prefer use `_x` instead of `x` and `_name` instead of `name` as macro arguments names. This prevent possible collisions with C variables labeled `x` or `name` visible in the scope where these macro are used.

Reply:
!! => removed (bad habit for pre C99 compiler recommendation)

for macro parameter: 'x' / 'name' => '_x' / '_name'
done even if I am surprised by this recommendation for a simple macro parameter
and it is not used in lot of existing code.....

for me, the "x" parameter can't have collision with local variable, 
because the local variable are parsed after pre-processing

I understood this need only for local variables defined when MACRO generate code

#define  INIT_ARRAY(tab) for (int i=0, _i <ARRAY_SIZE(tab); i++) { tab[i] = 0}

 int my_array[3];
 bool i = false;
 long tab[10];

 INIT_ARRAY(my_array) ;

=> this code cause conflict with variable in code generated with macro

in this case it is recommended to use "_i" instead of "i" for local variable (but no issue for parameter)

Reply:
My bad. You're right. I messed-up with some labels constraints on macro implementation.
My apologies. Thanks for the details.